### DISCUSSION TITLE
Question about the current state of WebGPU

### DISCUSSION THREAD

**Discussion by randspace0** [gfx-rs/wgpu#2163] [General]
I was trying [wgpu-rs](https://github.com/gfx-rs/wgpu) library and I want to try my program to run on the web with WebGPU for browser. But, my browser can't get the adapter, it always gives me `panicked at 'Failed to find an appropriate adapter'`. I look at some demos online such as https://toji.github.io/webgpu-test/ (1) and https://webkit.org/demos/webgpu/hello-triangle.html (2) and somehow, all the demos are errors.

Error on 1

```
renderer init failed TypeError: Cannot read properties of null (reading 'getSwapChainPreferredFormat')
    at WebGPURenderer.init (webgpu-renderer.js:200:41)
    at async onApiChange ((index):96:13)
```

Error on 2

```
Uncaught (in promise) TypeError: vertexBuffer.mapWriteAsync is not a function
    at helloTriangle (hello-triangle.js:60:50)
```

Is there any breaking change in the library, specs, or browser? Am I missing something? Is it related to my browser's inability to get the adapter?

## Step to reproduce

- Clone https://github.com/gfx-rs/wgpu/blob/master/wgpu/examples/hello-triangle/main.rs
- Change the code to initialize the instance with the WebGPU flag

```
let instance = wgpu::Instance::new(wgpu::Backends::BROWSER_WEBGPU);
```

## Browser:
- Firefox Nightly 96.0a1 (2021-11-06) (64-bit)
- Google Chrome Canary Version 98.0.4694.0 (Official Build) canary (64-bit)

**Comment by grovesNL** [comment]
Hi @andraantariksa! :wave: At the moment there are occasionally changes in the WebGPU specification which might temporarily break support for one of the browsers.

Until the specification stabilizes, we're focused on supporting Firefox Nightly first, so sometimes Chrome Canary might not work for some or all examples. The steps to running on Firefox Nightly are listed under https://github.com/gpuweb/gpuweb/wiki/Implementation-Status#firefox You can check if it's setup correctly in your browser by trying our own examples at https://wgpu.rs/examples/?example=hello-triangle which are usually up-to-date with Firefox Nightly.

To target WebGPU using wgpu you can follow the steps listed in our wiki at https://github.com/gfx-rs/wgpu/wiki/Running-on-the-Web-with-WebGPU-and-WebGL#webgpu This uses the same steps that we use to generate the wgpu.rs examples above.

**Reply by randspace0** [reply]
Thanks for the explanation. `gecko` branch fixed my problem 😀.
